### PR TITLE
Reworking a lot of components, polishing

### PR DESCRIPTION
**Pull request by SupremeSalvatore**

### CODE REVIEW

File: components/breadcrumb-item.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const BreadcrumbItem = {
+  template,
+  props: {
+    current: {
+      type: Boolean,
+      default: false
+    },
+    href: {
+      type: String,
+      default: '#'
```
Comment (alex-zendesk) [inline components/breadcrumb-item.js]: Please make this `javascript:void(0);` instead

File: components/breadcrumb-item.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const template = `
+<li class="c-breadcrumb__item" :class="classes">
+    <a v-if="!current" :href="href">
+      <slot></slot>
+    </a> 
+    <slot v-else></slot>
```
Comment (alex-zendesk) [inline components/breadcrumb-item.js]: Can you use `v-if` and `v-else` without using `<template>`?

File: components/callout.js [hunk 3]
```diff
@@ -35,12 +42,16 @@ const Callout = {
...
-        'c-callout--info': this.type === 'info'
+        'c-callout--dialog': this.dialog,
+        [`c-callout--${this.type}`]: this.type
       };
     }
+  },
+  methods: {
+    close() {
+      this.$destroy();
+      this.$el.parentNode.removeChild(this.$el);
```
Comment (alex-zendesk) [inline components/callout.js:53]: Is there a better way to remove this element? And how does this impact `v-if`, what if the call-out should appear again?

File: components/checkbox.js [hunk 4]
```diff
@@ -1,15 +1,34 @@
 const template = `
   <div class="c-chk">
-    <input class="c-chk__input" :id="name" type="checkbox" :checked="value" @change="$emit('input', $event.target.checked)" />
-    <label class="c-chk__label" :class="{'is-checked': value}" :for="name">
+    <input class="c-chk__input" :id="name" type="checkbox" :disabled="disabled===true" :checked="value" @change="change" />
```
Comment (alex-zendesk) [inline components/checkbox.js]: `disabled===true` let's stick to using truthy/falsey values in this case, you never know how the devs might be passing in these props

File: components/menu-item.js [hunk 5]
```diff
@@ -1,6 +1,15 @@
 const template = `
-  <li class="c-menu__item" role="menuitem" @click="$emit('click')">
-    {{ label }}
+  <li class="c-menu__item" :class="classes" role="menuitem" @click="$emit('click')">
+    <template v-if="separator===false">
```
Comment (alex-zendesk) [inline components/menu-item.js]: Change to falsey

File: components/menu.js [hunk 6]
```diff
@@ -1,9 +1,7 @@
 const template = `
-  <div class="menu-container u-1/1" v-show="open">
-    <ul class="c-menu" :class="classes" role="menu">
-      <slot></slot>
-    </ul>
-  </div>
+<ul class="c-menu" :class="classes" role="menu" :arria-hidden="open===false">
```
Comment (alex-zendesk) [inline components/menu.js]: Change to falsey

File: components/breadcrumb-item.js (hunk 1 above)
Comment (SupremeSalvatore) [reply components/breadcrumb-item.js]: Ok I see this makes sense

File: components/breadcrumb-item.js (hunk 2 above)
Comment (SupremeSalvatore) [reply components/breadcrumb-item.js]: I miss to see where am I using `<template>` ?

File: components/callout.js (hunk 3 above)
Comment (SupremeSalvatore) [reply components/callout.js:53]: We should talk about this, im not certain this is the best way, but it some kind of a way

File: components/checkbox.js (hunk 4 above)
Comment (SupremeSalvatore) [reply components/checkbox.js]: Thing is default html attributes afaik cant receive truthy/falsey value instead then need to have whole comparion, so we shoiuld probably make this as a computed if its even going to work. This answers other observations aswell

**Pull request closed without merging**